src/osdatahub/NGD/ngd_api.py

```diff
@@ -0,0 +1,212 @@
+            crs (str|int, optional): The CRS for the returned features, either in the format "epsg:xxxx" or an epsg
+                number. e.g. British National Grid can be supplied as "epsg:27700" or 27700.
+                Available CRS values are: EPSG:27700, EPSG:4326, EPSG:7405, EPSG:3857, and CRS84. Defaults to CRS84
+            start_datetime (datetime, optional): Selects features that have a temporal property after the given start
+                time
+            end_datetime (datetime, optional): Selects features that have a temporal property before the given end
+                time
+            cql_filter (str, optional): A filter query in the CQL format. More information about supported CQL operators
+                can be found at https://osdatahub.os.uk/docs/ofa/technicalSpecification
+            filter_crs (str|int, optional): The CRS for a given CQL query, either in the format "epsg:xxxx" or an epsg
+                number. e.g. British National Grid can be supplied as "epsg:27700" or 27700
+                Available CRS values are: EPSG:27700, EPSG:4326, EPSG:7405, EPSG:3857, and CRS84. Defaults to CRS84
+            max_results (int, optional): The maximum number of features to return. Defaults to 100
+            offset (int, optional): The offset number skips past the specified number of features in the collection.
+                Defaults to 0
+
+        Returns:
+            FeatureCollection: The results of the query in GeoJSON format
+        """
+
+        assert check_argument_types()
+        assert max_results > 0, f"Argument max_results must be greater than 0 but was {max_results}"
+        assert offset >= 0, f"Argument offset must be greater than 0 but was {offset}"
+        params = {}
+
+        # Checking validity and preformatting arguments
+        if crs:
+            params["crs"] = get_crs(crs=crs)
+
+        if start_datetime or end_datetime:
+            if start_datetime and end_datetime and start_datetime > end_datetime:
+                raise ValueError("Start time must be before end time")
+
+            start_datetime = start_datetime.isoformat() + "Z" if start_datetime else ".."
+            end_datetime = end_datetime.isoformat() + "Z" if end_datetime else ".."
+
+            params["datetime"] = f"{start_datetime}/{end_datetime}"
+
+        if extent:
+            # If extent is a bounding box, pass it as a bbox parameter
+            if extent.is_bbox:
+                params["bbox"] = extent.bbox.to_string(precision=5)
+                params["bbox-crs"] = get_crs(extent.crs, valid_crs=("epsg:4326", "epsg:27700", "epsg:3857", "crs84"))
+            # If extent is a polygon, implement spatial filter as an Intersects CQL filter
+            else:
+                bbox_filter = f"INTERSECTS(geometry, {extent.polygon.wkt})"
+
+                # ADD INTERSECTS QUERY
+                if cql_filter:
+                    if filter_crs:
```

Comment:
If Extent requires a crs, does filter_crs need to be an option?

Reply:
It is possible to make a request that has both a bbox and an INTERSECTS filter (e.g. a polygon), presumably this is possible with different crs for each but I have only tested on the same crs.